default.nix

```diff
@@ -122,6 +122,7 @@ lib.makeScope pkgs.newScope (self: {
       # Example: { my-app = ./src; }
     , editablePackageSources ? { }
     , __isBootstrap ? false  # Hack: Always add Poetry as a build input unless bootstrapping
+    , repositoryCredentialsFile ? null
```

Comment:
I don't really like the explicit passing of credentials. I would rather we use `impureEnvVars` https://nixos.org/manual/nix/stable/#sec-advanced-attributes or make the fetcher netrc aware.

Reply:
Thanks, I wasn't aware of those options.
I've updated it to use netrc, but it has to be passed in via both `extra-sandbox-paths` and an environment var to point to it. I struggled with `impureEnvVars` due to building via the daemon, so I grabbed the variable from the nix path as done in fetchdocker:
- https://github.com/NixOS/nixpkgs/blob/a0dbe47318bbab7559ffbfa7c4872a517833409f/pkgs/build-support/fetchdocker/credentials.nix

I can now install packages from a private pypi via `nix-shell -I NETRC=/etc/nix/.netrc --option extra-sandbox-paths /etc/nix
`, without explicitly passing credentials through the derivation.

Have I missed anything else basic?